### ISSUE TITLE
Submitting text to HM-Dis-EP-WM55 no longer possible

### ISSUE THREAD

**Issue by wtremmel** [zapccu/HMCCU#140]
Copied that from the forum so it is not forgotten...

Command:
`set Schlafzimmer_Taster datapoint 3.SUBMIT text1="09:00:00",text3="17:01:57"`

Error:
`HMCCUDEV: Schlafzimmer_Taster Usage: set Schlafzimmer_Taster datapoint [{channel-number}.]{datapoint} {value} [...]`

Context:
```
DEV Taster_Schlafzimmer NEQ1596198 interface=BidCos-RF type=HM-Dis-EP-WM55
CHN NEQ1596198:0 Taster_Schlafzimmer:0
   0.UNREACH = false {b} [RE]
   0.STICKY_UNREACH = true {b} [RWE]
   0.CONFIG_PENDING = false {b} [RE]
   0.LOWBAT = false {b} [RE]
   0.RSSI_DEVICE = 1 {n} [RE]
   0.RSSI_PEER = 1 {n} [RE]
   0.DEVICE_IN_BOOTLOADER = false {b} [RE]
   0.UPDATE_PENDING = false {b} [RE]
   0.AES_KEY = 0 {n} [R]
CHN NEQ1596198:1 HM-Dis-EP-WM55 NEQ1596198:1
   1.PRESS_SHORT = false {b} [WE]
   1.PRESS_LONG =  {b} [WE]
   1.INSTALL_TEST = false {b} [E]
   1.PRESS_CONT =  {b} [E]
   1.PRESS_LONG_RELEASE =  {b} [E]
CHN NEQ1596198:2 HM-Dis-EP-WM55 NEQ1596198:2
   2.PRESS_SHORT = false {b} [WE]
   2.PRESS_LONG = false {b} [WE]
   2.INSTALL_TEST = false {b} [E]
   2.PRESS_CONT = false {b} [E]
   2.PRESS_LONG_RELEASE = false {b} [E]
CHN NEQ1596198:3 HM-Dis-EP-WM55 NEQ1596198:3
   3.SUBMIT =  {s} [W]

Device detection:
StateDatapoint = 1.PRESS_SHORT [KEY]
StateDatapoint = 2.PRESS_SHORT [KEY]
ControlDatapoint = 1.PRESS_SHORT [KEY]
ControlDatapoint = 2.PRESS_SHORT [KEY]

Recommended module for device definition: HMCCUCHN

Current state datapoint = 1.PRESS_SHORT

Current control datapoint = 1.PRESS_SHORT

Device description

Device NEQ1596198 Taster_Schlafzimmer [HM-Dis-EP-WM55]
  CHILDREN: NEQ1596198:0,NEQ1596198:1,NEQ1596198:2,NEQ1596198:3,NEQ1596198:4,NEQ1596198:5,NEQ1596198:6,NEQ1596198:7,NEQ1596198:8
  FIRMWARE: 1.2
  FLAGS: Visible
  INTERFACE: MEQ1489693
  PARAMSETS: MASTER
  RF_ADDRESS: 5265659
  ROAMING: 0
  RX_MODE: BURST
  UPDATABLE: 1
Channel NEQ1596198:0 Taster_Schlafzimmer:0 [MAINTENANCE]
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible,Internal
  PARAMSETS: MASTER,VALUES
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:1 HM-Dis-EP-WM55 NEQ1596198:1 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: SENDER
  FLAGS: Visible
  GROUP: NEQ1596198:2
  LINK_SOURCE_ROLES: KEYMATIC,REMOTECONTROL_RECEIVER,SWITCH,WINMATIC
  PARAMSETS: LINK,MASTER,VALUES
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:2 HM-Dis-EP-WM55 NEQ1596198:2 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: SENDER
  FLAGS: Visible
  GROUP: NEQ1596198:1
  LINK_SOURCE_ROLES: KEYMATIC,REMOTECONTROL_RECEIVER,SWITCH,WINMATIC
  PARAMSETS: LINK,MASTER,VALUES
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:3 HM-Dis-EP-WM55 NEQ1596198:3 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER,VALUES
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:4 HM-Dis-EP-WM55 NEQ1596198:4 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:5 HM-Dis-EP-WM55 NEQ1596198:5 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:6 HM-Dis-EP-WM55 NEQ1596198:6 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:7 HM-Dis-EP-WM55 NEQ1596198:7 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55
Channel NEQ1596198:8 HM-Dis-EP-WM55 NEQ1596198:8 [KEY] known
  AES_ACTIVE: 0
  DIRECTION: NONE
  FLAGS: Visible
  PARAMSETS: MASTER
  PARENT: NEQ1596198
  PARENT_TYPE: HM-Dis-EP-WM55

Defaults

Support for role KEY of device type HM-Dis-EP-WM55 is built in.

```

**Comment by zapccu** [comment]
@wtremmel Please try this

set Schlafzimmer_Taster datapoint 3.SUBMIT="text1=09:00:00,text3=17:01:57"

Parsing of the command line is a bit weird (FHEM internal problem).

**Comment by wtremmel** [comment]
At least I no longer get an error…

but this should read:
set Schlafzimmer_Taster datapoint 3.SUBMIT=“text1:09-00-00,text3:17-01-57”

you changed the separator inside the string from “=“ to “:” (not that good when I want to display time values)
Can you change it back to “=“? (unless that messes up the outer parsing)

There is still some parsing error - I guess in the substr() call...

**Comment by wtremmel** [comment]
forget that - if I remove all punctuation from the string it works:

set Schlafzimmer_Taster datapoint 3.SUBMIT="text1:090000,text3:164628"

**Comment by zapccu** [comment]
Fixed it. I had to change the separator from "=" to ":" because of problems with the FHEM command line parser. In addition now only the first ":" is treated as a separator. After next update this will work:

set HM_EPDISP datapoint 3.SUBMIT "text1:Line2,text2:Has Blank,text3:10:05:21,sound:snd_short,signal:sig_red

Hint: The epaper display must be defined by using module HMCCUDEV with option "forceDev":

define myDev HMCCUDEV addressOrName forceDev
